connect-inject/health_check_resource.go

```diff
@@ -232,6 +234,13 @@ func (h *HealthCheckResource) getServiceCheck(client *api.Client, healthCheckID
 // ready state of the pod along with the reason message which will be passed into the Notes
 // field of the Consul health check.
 func (h *HealthCheckResource) getReadyStatusAndReason(pod *corev1.Pod) (string, string, error) {
+	// A pod might be pending if the init containers have run but the non-init
+	// containers haven't been started. In this case we set a failing health
```

Comment:
```suggestion
	// containers haven't reached running state. In this case we set a failing health
```
I'm not entirely sure how to re-word this but I think we need to instead state that we're going after pods that have not reached PodRunning, the containers have likely "started" but just **haven't reached the point where readiness probes have run.**

Reply:
I think your re-word is good.